### PR TITLE
CORE-13740: Change the virtual node database connection pool to have 0 idle connections

### PR DESCRIPTION
**Pull request by yift-r3**
Make the virtual node DML database connection pool have more than one connection and set the minimal number of connections to 0.

The API pull request is in https://github.com/corda/corda-api/pull/1128

This is the first pull request of two; the second will use this function in the members and certificate persistence service. - see https://github.com/corda/corda-runtime-os/pull/3916

### CODE REVIEW

File: components/db/db-connection-manager/src/main/kotlin/net/corda/db/connection/manager/DbConfig.kt [hunk 1]
```diff
@@ -34,6 +34,14 @@ fun DataSourceFactory.createFromConfig(config: SmartConfig): CloseableDataSource
     val driver = configWithFallback.getString(DatabaseConfig.JDBC_DRIVER)
     val jdbcUrl = configWithFallback.getString(DatabaseConfig.JDBC_URL)
     val maxPoolSize = configWithFallback.getInt(DatabaseConfig.DB_POOL_MAX_SIZE)
+    val minPoolSize = if(
+        configWithFallback.hasPath(DatabaseConfig.DB_POOL_MIN_SIZE)
+        && !configWithFallback.getIsNull(DatabaseConfig.DB_POOL_MIN_SIZE)
```
Comment (fowlerrr) [inline components/db/db-connection-manager/src/main/kotlin/net/corda/db/connection/manager/DbConfig.kt]: `hasPath()` already checks for `null` so you don't need the second bit, I think.

Comment (yift-r3) [reply components/db/db-connection-manager/src/main/kotlin/net/corda/db/connection/manager/DbConfig.kt]: Yes, that is correct.

File: libs/db/db-core/src/main/kotlin/net/corda/db/core/HikariDataSourceFactory.kt [hunk 2]
```diff
@@ -66,8 +66,14 @@ class HikariDataSourceFactory(
         conf.isAutoCommit = isAutoCommit
         conf.isReadOnly = isReadOnly
         conf.maximumPoolSize = maximumPoolSize
-        conf.minimumIdle = minimumPoolSize
-        conf.idleTimeout = idleTimeout.toMillis()
+        if (minimumPoolSize != null) {
+            conf.minimumIdle = minimumPoolSize
+        }
```
Comment (kyriathar) [inline libs/db/db-core/src/main/kotlin/net/corda/db/core/HikariDataSourceFactory.kt]: Shouldn't we say that if minimumPoolSize == null then `conf.minimumIdle = maximumPoolSize`?

Comment (yift-r3) [reply libs/db/db-core/src/main/kotlin/net/corda/db/core/HikariDataSourceFactory.kt]: If it's null, we can leave it as is and let Hikari manage the default.

**Review by kyriathar — DISMISSED** [review]

**Review by kyriathar — APPROVED** [review]

**Review by dimosr — APPROVED** [review]

**Pull request merged**